pkg/cmd/server/handlers/impersonation.go

```diff
@@ -56,19 +56,22 @@ func ImpersonationFilter(handler http.Handler, a authorizer.Authorizer, groupCac
 			}
 
 			switch subject.GetObjectKind().GroupVersionKind().GroupKind() {
-			case userapi.Kind(authorizationapi.GroupKind):
+			case userapi.Kind(authorizationapi.GroupKind),
+				userapi.LegacyKind(authorizationapi.GroupKind):
```

Comment:
Can you leave this line in one line in case? It reads soooo much better. I got confused by looking at it first.